doc/ref_arch/kubernetes/chapters/chapter02.md

```diff
@@ -20,20 +20,148 @@
+| `req.gen.cnt.02` | Cloud nativeness | The Architecture **should** consist of service components implemented as microservices that are individually dynamically scalable. |
+| `req.gen.cnt.03` | Cloud nativeness | The Architecture **must** support immutable infrastructure. |
+| `req.gen.scl.01` | Scalability | The Architecture **should** support policy driven horizontal auto-scaling of workloads. |
+| `req.gen.rsl.01` | Resiliency | The Architecture **must** support resilient Kubernetes components that are required for the continued availability of running workloads. |
+| `req.gen.rsl.02` | Resiliency | The Architecture **should** support resilient Kubernetes service components that are not subject to `req.gen.rsl.01`. |
+| `req.gen.avl.01` | Availability | The Architecture **must** provide High Availability for Kubernetes components. |
+| `req.gen.ref.01` | Model | The Architecture **must** support the Reference Model defined profiles (instance types: Basic, Network Intensive, Compute Intensive). |
+
+
+<p align="center"><b>Table 2-1:</b> Kubernetes Architecture: General Requirements</p>
+
+<a name="2.3.2"></a>
+### 2.3.2 Infrastructure
+
+| Ref # | sub-category | Description |
+|----|------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `req.inf.com.01` | Compute | The Architecture **must** provide compute resources for Pods.  |
+| `req.inf.com.02` | Compute | The Architecture **should** include industry standard hardware management systems at both HW device level (embedded) and HW platform level (external to device). |
+| `req.inf.com.03` | Network | The Architecture **must** support Conatiner Runtime Interface (CRI). |
+| `req.inf.stg.01` | Storage | The Architecture **must**  support for Kubernetes Volumes for container storage.
+| `req.inf.stg.02` | Storage | The Architecture **may** provide shared Object storage as a service for Containers workload.
+| `req.inf.stg.03` | Storage | The Architecture **must** support Container Storage Interfaces (CSI).
+| `req.inf.stg.04` | Storage | The Architecture **may** support Software Defined Storage (SDS) that seamlessly supports shared block storage, object storage and flat files. |
+| `req.inf.stg.05` | Storage | The Architecture **should** provide high-performance and horizontally scalable storage. |
+| `req.inf.ntw.01` | Network | The Architecture **must** support intra-pod communications such as between all containers within a pod on localhost |
+| `req.inf.ntw.02` | Network | The Architecture **must** support intra-node communications, such as between agents on a node and all pods on that node |
+| `req.inf.ntw.03` | Network | The Architecture **must** support inter-node communications without NAT, such as communications between pods on a node with all other pods on all nodes |
```

Comment:
This is again inherently provided by K8s. Do we need to specify it explicitly ?

Reply:
for the previous one your remark is true, but this one is not provided by K8s out-of-box
it depends on CNI

Reply:
I think CNI plugins have a mandate that below communications should be without NAT
1. pod-to-pod
2.pod-to-node
3.node-to-node

Reply:
not at all
bridge CNI, Flannel CNI etc.

Reply:
My understanding is from this [link](https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model)

Reply:
yeah I figured :)
often quoted, but the truth is that this whole chapter is entirely not true. it is actually originating from times where kubenet was the only networking solution.
Kubernetes foregone the right to state  any requirements towards network fabric after the introduction of CNI

I'll give you an example from our own cloud. I have Flannel, which is a VxLAN overlay, and a bridge. I also have some CaaS services running in hostns, namely DNS, API server itself etc.
I could easily setup routing rules so that Pods having Flannel IPs can still reach these infra services inside the cluster, but it doesn't mean Flannel network is not an overlay network.

Reply:
the only thing Kubernetes imposes on a CNI plugin is the implementation of the spec, i.e. CNI_ADD, CNI_DEL, and recently CNI_GET

Reply:
Hmm, I see :). So you would like to keep req.inf.ntw.03 as is ?

Reply:
summarizing: this requirement IMO is very much needed, because the quoted Flannel network interface is unsuitable for data plane communication (even for kernel networking VNFs)

Reply:
Ok, thanks for sharing your thoughts :).